### ISSUE TITLE
Generate dynamic password length

### ISSUE THREAD

**Issue by zpcol** [Kunzisoft/KeePassDX#699]
generate dynamic password length

**Comment by J-Jamet** [comment]
Can you be more specific, I don't quite understand the request.

**Comment by zpcol** [comment]
in menu App-Setting - Generate Password Size

could you add a non-static size option, for example 10-20 bits. 8-30 bits, etc. every time you click "generate password".

**Comment by zpcol** [comment]
resulting in a different password length for each click.

example, the first click produces 21 length, second click 13 length, third click 17 length password, etc.

**Comment by zpcol** [comment]
generate not only random code, but also a different number of characters / bits.

**Comment by J-Jamet** [comment]
OK, I understand better, why not, I will see when refactoring the password generator.

**Comment by J-Jamet** [comment]
I reopen the issue, otherwise I do not know what has already been done. ;)

**Comment by zpcol** [comment]
I also searched duckduckgo there is no web password generator online that has random length feature.. :)

**Comment by K4zoku** [comment]
If it having a bound seems to be greater (e.g: 8-32 characters)

**Comment by Pflanzenritter29** [comment]
Heya 👋

I have some critique on this;

TL;DR:
I don't think having password length as randomness, will gain you any advantage, compared to how much you will lose when getting a shorter amount of characters.


_From my understanding, it is always the better option - entropy-wise - to go with the higher number of characters.
The entropy-calculation-algo is only really useful for estimating how good a password, that is not entirely randomly generated, is, and will therefore still fluctuate somewhat, even with a fixed amount of characters and a completely randomly generated one.

While having the option to randomly generated a password length from 8-32 for example, may sound intuitive, it would make your password offline-crackable relatively easily when it gets below 10 characters.

Just think of it like this: There is a reason, why we shouldn't gain pretty much any advantage to cracking a randomly generated password with enough characters, just by knowing the "type" (e.g. passphrase or password), or length.

It also should always be reduced to just the secret/information, and not the "enviromental" conditions be used as a way to add to entropy/security.
You should always assume an attacker knows the enviroment, and then still make it secure.

(Having 5-7 word passphrases for example; having a 7 word one vs randomly getting a 5 word one, isn't really different from just using a 5 word one in the first place - I'm pretty sure you essentially gain nothing from that 1/3 chance, compared to an additional ^2 )_

**Comment by Pflanzenritter29** [comment]
@J-Jamet Am I missing something fundamental here, or would you agree?

(Also I am quite new to contributing on Github like this, so excuse me if I'm missing some etiquette)

**Comment by J-Jamet** [comment]
I agree with what you say. (I simply hadn't reacted)
